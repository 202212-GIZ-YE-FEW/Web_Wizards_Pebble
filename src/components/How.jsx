import Image from "next/image";
import Link from "next/link";

import { BorderlessButton } from "@/components/mini";
import { Title } from "@/components/mini";

import attendImage from "~/landing/attend.svg";
import boyImage from "~/landing/boy.svg";
import girlImage from "~/landing/girl.svg";

function Cards(props) {
    const { t } = props;
    return (
        <div className='grid grid-cols-1 lg:grid-cols-3 gap-y-20 lg:gap-x-10'>
            <Card
                title={`${t("registerTitle")}`}
                description={`${t("registerDescription")}`}
                img={boyImage}
                height={237.71}
                width={274}
                alt='Boy with notebook.'
                buttonText={`${t("registerButton")}`}
                buttonTo='/signup'
            />
            <Card
                title={`${t("attendTitle")}`}
                description={`${t("attendDescription")}`}
                img={attendImage}
                height={326.03}
                width={236.42}
                alt='People in an event.'
                buttonText={`${t("attendButton")}`}
                buttonTo='/events'
            />
            <Card
                title={`${t("organizeTitle")}`}
                description={`${t("organizeDescription")}`}
                img={girlImage}
                height={245.75}
                width={216}
                alt='Girl with a speaker'
                buttonText={`${t("organizeButton")}`}
                buttonTo='/'
            />
        </div>
    );
}

function Card(props) {
    const {
        title,
        description,
        img,
        alt,
        height,
        width,
        buttonText,
        buttonTo,
    } = props;
    return (
        <div className='flex flex-col items-center text-center'>
            <Image
                src={img}
                alt={alt}
                width={height}
                height={width}
                className='h-full'
            />
            <h3 className='text-black-100 font-medium mt-2 text-3xl font-sans'>
                {title}
            </h3>
            <p className='text-black-50 text-xl leading-8 !my-4'>
                {description}
            </p>
            <Link href={buttonTo}>
                <BorderlessButton>{buttonText}</BorderlessButton>
            </Link>
        </div>
    );
}

export default function How(props) {
    const { t } = props;
    return (
        <section className='container mt-40'>
            <Title className='pb-32'>{t("howItWorks")}</Title>
            <Cards t={t} />
        </section>
    );
}
