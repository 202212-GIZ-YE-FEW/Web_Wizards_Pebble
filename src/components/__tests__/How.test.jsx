import renderer from "react-test-renderer";

import How from "@/components/How";

it("renders correctly", () => {
    const tree = renderer.create(<How t={(str) => str} />).toJSON();
    expect(tree).toMatchSnapshot();
});
