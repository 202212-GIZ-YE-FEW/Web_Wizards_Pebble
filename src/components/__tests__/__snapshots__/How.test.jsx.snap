// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="container mt-40"
  id="how_it_works"
>
  <h2
    className="text-center text-black-100 text-4xl font-sans pb-32"
  >
    howItWorks
  </h2>
  <div
    className="grid grid-cols-1 lg:grid-cols-3 gap-y-20 lg:gap-x-10"
  >
    <div
      className="flex flex-col items-center text-center"
    >
      <img
        alt="Boy with notebook."
        className="h-full"
        data-nimg="1"
        decoding="async"
        height={274}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={237.71}
      />
      <h3
        className="text-black-100 font-medium mt-2 text-3xl font-sans"
      >
        registerTitle
      </h3>
      <p
        className="text-black-50 text-xl leading-8 !my-4"
      >
        registerDescription
      </p>
      <a
        href="/signup"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="button__ButtonWrapper-sc-1aw98b4-0 button !rounded-none !text-black-100 !bg-transparent !shadow-none !border-none after:content-[url(\\"/arrow.svg\\")] after:mt-[1px] ltr:after:ml-4 after:absolute rtl:after:rotate-180 rtl:after:mr-4"
          disabled={false}
        >
          registerButton
        </button>
      </a>
    </div>
    <div
      className="flex flex-col items-center text-center"
    >
      <img
        alt="People in an event."
        className="h-full"
        data-nimg="1"
        decoding="async"
        height={236.42}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=750&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=750&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={326.03}
      />
      <h3
        className="text-black-100 font-medium mt-2 text-3xl font-sans"
      >
        attendTitle
      </h3>
      <p
        className="text-black-50 text-xl leading-8 !my-4"
      >
        attendDescription
      </p>
      <a
        href="/events"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="button__ButtonWrapper-sc-1aw98b4-0 button !rounded-none !text-black-100 !bg-transparent !shadow-none !border-none after:content-[url(\\"/arrow.svg\\")] after:mt-[1px] ltr:after:ml-4 after:absolute rtl:after:rotate-180 rtl:after:mr-4"
          disabled={false}
        >
          attendButton
        </button>
      </a>
    </div>
    <div
      className="flex flex-col items-center text-center"
    >
      <img
        alt="Girl with a speaker"
        className="h-full"
        data-nimg="1"
        decoding="async"
        height={216}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={245.75}
      />
      <h3
        className="text-black-100 font-medium mt-2 text-3xl font-sans"
      >
        organizeTitle
      </h3>
      <p
        className="text-black-50 text-xl leading-8 !my-4"
      >
        organizeDescription
      </p>
      <a
        href="/events/create"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="button__ButtonWrapper-sc-1aw98b4-0 button !rounded-none !text-black-100 !bg-transparent !shadow-none !border-none after:content-[url(\\"/arrow.svg\\")] after:mt-[1px] ltr:after:ml-4 after:absolute rtl:after:rotate-180 rtl:after:mr-4"
          disabled={false}
        >
          organizeButton
        </button>
      </a>
    </div>
  </div>
</section>
`;
